sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereQuery.cs

```diff
@@ -92,13 +103,37 @@ public AdtQuerySelectFrom WhereIsOfModel(string model, bool exact = false)
         /// Adds the logical operator AND to the query.
         /// </summary>
         /// <returns> ADT query that already contains SELECT and FROM. </returns>
-        public AdtQuerySelectFrom And()
+        public WhereQuery And()
         {
             return this;
         }
 
+        /// <inheritdoc/>
+        public override AdtQueryBuilder Build()
+        {
+            return _parent;
+        }
+
+        /// <inheritdoc/>
+        public override string Stringify()
+        {
+            // TODO -- make this cleaner? design problem?
+            if (_clauses.Any())
+            {
+                StringBuilder whereComponents = new StringBuilder();
```

Comment:
general rule of thumb is if you are concatenating 3 or more strings together to instead use a string builder.

Right now you just have two, and you could even use string interpolation which would make it even more readable (probably).

Unless you are anticipating adding more strings, I'd advise against using a string builder.